src/lib/routes/admin-api/user-admin.ts

```diff
@@ -716,4 +725,28 @@ export default class UserAdminController extends Controller {
             projectRoles,
         });
     }
+
+    async throwIfScimUser({
+        id,
+        scimId,
+    }: Pick<IUser, 'id' | 'scimId'>): Promise<void> {
+        if (this.isEnterprise && this.flagResolver.isEnabled('scimApi')) {
```

Comment:
Gosh that's a lot of nested ifs! How would you feel about flattening this one a little?

I'm also tempted to move this check:

```
                const isScim =
                    Boolean(scimId)
```

Up in the function call. That looks super cheap. It'd be cool if we can early exit without making an API call!

Reply:
Agreed, not a fan either. Tried addressing it here: https://github.com/Unleash/unleash/pull/6836/commits/9b88eb26e08bfae867c52ac4b0b8a50360e84cb5

Reply:
Yeah that's better!